code/src/reducers/labyrinth.js

```diff
@@ -0,0 +1,95 @@
+/*eslint-disable*/
+import { createSlice } from '@reduxjs/toolkit';
+import loading from './loading';
+
+const labyrinth = createSlice ({
+  name: 'labyrinth',
+  initialState: {
+    username: '',
+    coordinates:'', 
+    description: '',
+    actions: []
+  },
+
+    reducers: {
+      setupStore: (store, action) => {
+        store.items = action.payload;
+      },
+      setUserName: (state, action) => {
+        state.username = action.payload
+      },
```

Comment:
This code below would allow the project to work even when using same user name without bugs. 
```
setUserName: (store, action) => {
      store.username = `${new Date().getTime()} ${action.payload}`
```